### Original Dataframe has been split into 2 categories based on Indicator type to make 2 smaller dataframes:

# ** 1. INFORM INDEX
# 2. Core Indicators

## Goal: Split the INFORM Index dataframes further based on Indicator ID column categories

### Importing 10 dataframes for INFORM Index

In [1]:
import pandas as pd

In [2]:
df1 = pd.read_csv("INFORM Index 2015.csv")

In [3]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32541 entries, 0 to 32540
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32541 non-null  object 
 1   IndicatorId     32541 non-null  object 
 2   IndicatorName   32422 non-null  object 
 3   IndicatorScore  32541 non-null  float64
 4   SurveyYear      32541 non-null  int64  
 5   Indicator Type  32541 non-null  object 
 6   INFORMYear      32541 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [4]:
df1.head()

Iso3 IndicatorId                                 IndicatorName  \
0  AFG  AFF_DR_REL  People affected by droughts (relative) - raw   
1  AGO  AFF_DR_REL  People affected by droughts (relative) - raw   
2  ALB  AFF_DR_REL  People affected by droughts (relative) - raw   
3  ARE  AFF_DR_REL  People affected by droughts (relative) - raw   
4  ARG  AFF_DR_REL  People affected by droughts (relative) - raw   

   IndicatorScore  SurveyYear Indicator Type  INFORMYear  
0        0.568526           0    INORM Index        2015  
1        0.468029           0    INORM Index        2015  
2        3.170088           0    INORM Index        2015  
3        0.000000           0    INORM Index        2015  
4        0.000000           0    INORM Index        2015

In [5]:
df2 = pd.read_csv("INFORM Index 2016.csv")

In [6]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32566 entries, 0 to 32565
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32566 non-null  object 
 1   IndicatorId     32566 non-null  object 
 2   IndicatorName   32445 non-null  object 
 3   IndicatorScore  32566 non-null  float64
 4   SurveyYear      32566 non-null  int64  
 5   Indicator Type  32566 non-null  object 
 6   INFORMYear      32566 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [7]:
df2.head()

Iso3 IndicatorId                                 IndicatorName  \
0  AFG  AFF_DR_REL  People affected by droughts (relative) - raw   
1  AGO  AFF_DR_REL  People affected by droughts (relative) - raw   
2  ALB  AFF_DR_REL  People affected by droughts (relative) - raw   
3  ARE  AFF_DR_REL  People affected by droughts (relative) - raw   
4  ARG  AFF_DR_REL  People affected by droughts (relative) - raw   

   IndicatorScore  SurveyYear Indicator Type  INFORMYear  
0        0.551054           0    INORM Index        2016  
1        0.451400           0    INORM Index        2016  
2        3.171872           0    INORM Index        2016  
3        0.000000           0    INORM Index        2016  
4        0.000000           0    INORM Index        2016

In [8]:
df3 = pd.read_csv("INFORM Index 2017.csv")
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32645 entries, 0 to 32644
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32645 non-null  object 
 1   IndicatorId     32645 non-null  object 
 2   IndicatorName   32525 non-null  object 
 3   IndicatorScore  32645 non-null  float64
 4   SurveyYear      32645 non-null  int64  
 5   Indicator Type  32645 non-null  object 
 6   INFORMYear      32645 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [9]:
df4 = pd.read_csv("INFORM Index 2018.csv")
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32649 entries, 0 to 32648
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32649 non-null  object 
 1   IndicatorId     32649 non-null  object 
 2   IndicatorName   32537 non-null  object 
 3   IndicatorScore  32649 non-null  float64
 4   SurveyYear      32649 non-null  int64  
 5   Indicator Type  32649 non-null  object 
 6   INFORMYear      32649 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [10]:
df5 = pd.read_csv("INFORM Index 2019.csv")
df5.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32669 entries, 0 to 32668
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32669 non-null  object 
 1   IndicatorId     32669 non-null  object 
 2   IndicatorName   32555 non-null  object 
 3   IndicatorScore  32669 non-null  float64
 4   SurveyYear      32669 non-null  int64  
 5   Indicator Type  32669 non-null  object 
 6   INFORMYear      32669 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [11]:
df6 = pd.read_csv("INFORM Index 2020.csv")
df6.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32634 entries, 0 to 32633
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32634 non-null  object 
 1   IndicatorId     32634 non-null  object 
 2   IndicatorName   32517 non-null  object 
 3   IndicatorScore  32634 non-null  float64
 4   SurveyYear      32634 non-null  int64  
 5   Indicator Type  32634 non-null  object 
 6   INFORMYear      32634 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [12]:
df7 = pd.read_csv("INFORM Index 2021.csv")
df7.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32631 entries, 0 to 32630
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32631 non-null  object 
 1   IndicatorId     32631 non-null  object 
 2   IndicatorName   32518 non-null  object 
 3   IndicatorScore  32631 non-null  float64
 4   SurveyYear      32631 non-null  int64  
 5   Indicator Type  32631 non-null  object 
 6   INFORMYear      32631 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [13]:
df8 = pd.read_csv("INFORM Index 2022.csv")
df8.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32647 entries, 0 to 32646
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32647 non-null  object 
 1   IndicatorId     32647 non-null  object 
 2   IndicatorName   32533 non-null  object 
 3   IndicatorScore  32647 non-null  float64
 4   SurveyYear      32647 non-null  int64  
 5   Indicator Type  32647 non-null  object 
 6   INFORMYear      32647 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [14]:
df9 = pd.read_csv("INFORM Index 2023.csv")
df9.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32636 entries, 0 to 32635
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32636 non-null  object 
 1   IndicatorId     32636 non-null  object 
 2   IndicatorName   32522 non-null  object 
 3   IndicatorScore  32636 non-null  float64
 4   SurveyYear      32636 non-null  int64  
 5   Indicator Type  32636 non-null  object 
 6   INFORMYear      32636 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [15]:
df10 = pd.read_csv("INFORM Index 2024.csv")
df10.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32630 entries, 0 to 32629
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            32630 non-null  object 
 1   IndicatorId     32630 non-null  object 
 2   IndicatorName   32517 non-null  object 
 3   IndicatorScore  32630 non-null  float64
 4   SurveyYear      32630 non-null  int64  
 5   Indicator Type  32630 non-null  object 
 6   INFORMYear      32630 non-null  int64  
dtypes: float64(1), int64(2), object(4)
memory usage: 1.7+ MB


In [16]:
str(len(df1['IndicatorName'].unique()))

'174'

In [17]:
str(len(df1['IndicatorId'].unique()))

'186'

### Consider splitting each dataframe on IndicatorName  
- 174 categories in IndicatorName x 10 dataframes
- Store as (174 x 2? but names may differ in Core Indicators dataframe) 348 new dataframes for INFORM Index & Core Indicators - a lot of dataframes!!
- Keep original dataframes
- Export new worksheet for each year dataframes to Project folder

In [1]:
# Code for the split 
# Grouping by 'IndicatorId'
#grouped_df1 = df1.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
#df1s = {}

# Splitting and storing DataFrames based on category
#for category, group_df1 in grouped_df1:
#    df1s[category] = group_df1.copy()

# Accessing the new DataFrames and saving them
#for category, new_df1 in df1s.items():
#    print(f"DataFrame of 2015 Indeces for Indicator Id {category}:")
#    print(new_df1.head())
#    print()  # line to space between dataframes
#    print(new_df1.info())
#    print()  # line to space between dataframes
#    print()  # line to space between dataframes

##### 348 Dataframes for 2015-2024 Yearly INFORM Index Databases/Dataframes based on IndicatorNames created & stored remotely  - too many dataframes!!!

In [19]:
# Code for storing 348 dataframes remotely
#import os

# Define the folder path where you want to save the DataFrames
#folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
#for category, new_df1 in df1s.items():
    # Construct the file path
#    file_path = os.path.join(folder_path, f"2015 INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
#    new_df1.to_csv(file_path, index=False)
#    print(f"DataFrame for 2015 INFORM Year Indeces for {category} saved as {file_path}")

#### Need to reduce to selected indeces for each year, eg. INFORM Index, Main 3 component indeces + others...
#### In INFORM Index dataframe, 174 Indicator Names, 186 Indicator IDs

In [20]:
df1['IndicatorName'].unique().tolist()

['People affected by droughts (relative) - raw',
 'Health expenditure per capita',
 'Diphtheria-Tetanus-Pertussis',
 'Measles',
 'Pneumococcal',
 'Maternal Mortality Ratio',
 'Physicians density',
 'Mobile cellular subscriptions',
 'Access to electricity',
 'Adult literacy rate',
 'Internet users',
 'Access to improved water source',
 'Access to improved sanitation facilities',
 'Disaster Risk Reduction',
 'Corruption Perception Index',
 'Government Effectiveness',
 'Physical exposure to coastal flood - average annual population exposed (percentage of the total population) - raw',
 'Physical exposure to earthquake MMI VI (relative) - raw',
 'Physical exposure to earthquake MMI VIII (relative) - raw',
 'Physical exposure to river flood (relative) - raw',
 'Physical exposure to tropical cyclone of Saffir-Simpson category 1 (relative) - raw',
 'Physical exposure to tropical cyclone of Saffir-Simpson category 3 (relative) - raw',
 'Physical exposure to tsunami (relative) - raw',
 'IHR capa

In [21]:
sorted([str(value) for value in df1['IndicatorName'].unique().tolist()])

['Access to Health Care',
 'Access to electricity',
 'Access to improved sanitation facilities',
 'Access to improved water source',
 'Adult literacy rate',
 'Adult prevalence of HIV-AIDS',
 'Agriculture Drought Probability',
 'Children Underweight',
 'Communication',
 'Corruption Perception Index',
 'Current National Power conflicts',
 'Current Subnational conflicts',
 'Current conflicts',
 'Diphtheria-Tetanus-Pertussis',
 'Disaster Risk Reduction',
 'Droughts probability and historical impact',
 'Economic Dependency',
 'Estimated HDI from GDP per capita',
 'FTS %YEARREF|-1%-%YEARREF|-3%',
 'FTS / Capita',
 'Food',
 'Food Availability',
 'Food Security',
 'Food Utilization',
 'Frequency of drought events',
 'Gender Inequality Index',
 'Governance',
 'Government Effectiveness',
 'HIV',
 'HIV incidence',
 'Hazard & Exposure Index',
 'Hazard & Exposure Index - Epidemic',
 'Health Conditions',
 'Health expenditure per capita',
 'Health of children under-five',
 'Household size',
 'Human D

 #### Selected Indeces
1. 'INFORM Risk Index'                      - INFORM 

2. 'Hazard & Exposure Index',               - HA
  - 'Human Hazard',                           - HA.HUM
  - 'Current Conflicts',                      - HA.HUM.CON
  - 'Hazard & Exposure Index - Epidemic',     - HA.NAT.EPI
  - 'Violent Conflict probability Score',     - HA.HUM.CON.GCRI
  - 'Current Subnational conflicts',          - HA.HUM.CON.SN
  - 'Current National Power conflicts'??

3. 'Vulnerability Index',                   - VU
  - 'Socio-Economic Vulnerability',           - VU.SEV
  - 'Vulnerable Groups',                      - VU.VGR
  - 'Poverty & Development'                   - VU.SEV.PD
  - 'Inequality'                              - VU.SEV.INQ
  - 'Human Development Index',                - VU.SEV.PD.HDI
  - 'Multidimensional Poverty Index'          - VU.SEV.PD.MPI

4. 'Lack of Coping Capacity Index',         - CC
  - 'Infrastructure',                         - CC.INF
  - 'Communication'                           - CC.INF.COM	
  - 'Access to Health Care',                  - CC.INF.AHC
  - 'Physical Infrastructure',                - CC.INF.PHY
  - 'Governance',                             - CC.INS.GOV
  
 - ####  Total 19 Indeces Selected

### New Index Databases/Dataframes created & stored remotely for each Index Year 2015-2024

#### Reduced to 11 Indeces Selected
1. 'INFORM Risk Index'                      - INFORM 

2. 'Hazard & Exposure Index',               - HA
  - 'Human Hazard',                           - HA.HUM
  - 'Current Conflicts',                      - HA.HUM.CON
  - 'Hazard & Exposure Index - Epidemic',     - HA.NAT.EPI

3. 'Vulnerability Index',                   - VU
  - 'Socio-Economic Vulnerability',           - VU.SEV
  - 'Vulnerable Groups',                      - VU.VGR

4. 'Lack of Coping Capacity Index',         - CC
  - 'Infrastructure',                         - CC.INF
  - 'Governance',                             - CC.INS.GOV

### 2015 INFORM Indeces

In [22]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df1 = df1.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df1s = {}

# Splitting and storing DataFrames based on category
for category, group_df1 in grouped_df1:
    if category in selected_indicator_ids:
        df1s[category] = group_df1.copy()

# Accessing the new DataFrames and printing info
for category, new_df1 in df1s.items():
    print(f"DataFrame of 2015 Indices for Indicator Id {category}:")
    print(new_df1.head())
    print()  # line to space between dataframes
    print(new_df1.info())
    print()  # line to space between dataframes

DataFrame of 2015 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31395  AFG          CC  Lack of Coping Capacity Index             7.6   
31396  AGO          CC  Lack of Coping Capacity Index             7.1   
31397  ARE          CC  Lack of Coping Capacity Index             1.9   
31398  ARG          CC  Lack of Coping Capacity Index             3.6   
31399  ARM          CC  Lack of Coping Capacity Index             4.8   

       SurveyYear Indicator Type  INFORMYear  
31395           0    INORM Index        2015  
31396           0    INORM Index        2015  
31397           0    INORM Index        2015  
31398           0    INORM Index        2015  
31399           0    INORM Index        2015  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31395 to 31585
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [23]:
import os

# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df1 in df1s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2015 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df1.to_csv(file_path, index=False)
    print(f"DataFrame for 2015 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2015 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2015 Select INFORM Index - CC.csv
DataFrame for 2015 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2015 Select INFORM Index - CC.INF.csv
DataFrame for 2015 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2015 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2015 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2015 Select INFORM Index - HA.csv
DataFrame for 2015 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2015 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2015 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\

### 2016 INFORM Indeces

In [24]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df2 = df2.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df2s = {}

# Splitting and storing DataFrames based on category
for category, group_df2 in grouped_df2:
    if category in selected_indicator_ids:
        df2s[category] = group_df2.copy()

# Accessing the new DataFrames and printing info
for category, new_df2 in df2s.items():
    print(f"DataFrame of 2016 Indices for Indicator Id {category}:")
    print(new_df2.head())
    print()  # line to space between dataframes
    print(new_df2.info())
    print()  # line to space between dataframes

DataFrame of 2016 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31420  AFG          CC  Lack of Coping Capacity Index             7.6   
31421  AGO          CC  Lack of Coping Capacity Index             7.1   
31422  ARE          CC  Lack of Coping Capacity Index             1.8   
31423  ARG          CC  Lack of Coping Capacity Index             3.6   
31424  ARM          CC  Lack of Coping Capacity Index             4.8   

       SurveyYear Indicator Type  INFORMYear  
31420           0    INORM Index        2016  
31421           0    INORM Index        2016  
31422           0    INORM Index        2016  
31423           0    INORM Index        2016  
31424           0    INORM Index        2016  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31420 to 31610
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [25]:
# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df2 in df2s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2016 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df2.to_csv(file_path, index=False)
    print(f"DataFrame for 2016 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2016 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2016 Select INFORM Index - CC.csv
DataFrame for 2016 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2016 Select INFORM Index - CC.INF.csv
DataFrame for 2016 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2016 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2016 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2016 Select INFORM Index - HA.csv
DataFrame for 2016 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2016 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2016 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\

### 2017 INFORM Indeces

In [26]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df3 = df3.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df3s = {}

# Splitting and storing DataFrames based on category
for category, group_df3 in grouped_df3:
    if category in selected_indicator_ids:
        df3s[category] = group_df3.copy()

# Accessing the new DataFrames and printing info
for category, new_df3 in df3s.items():
    print(f"DataFrame of 2017 Indices for Indicator Id {category}:")
    print(new_df3.head())
    print()  # line to space between dataframes
    print(new_df3.info())
    print()  # line to space between dataframes

DataFrame of 2017 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31499  AFG          CC  Lack of Coping Capacity Index             7.4   
31500  AGO          CC  Lack of Coping Capacity Index             7.0   
31501  ARE          CC  Lack of Coping Capacity Index             1.9   
31502  ARG          CC  Lack of Coping Capacity Index             3.5   
31503  ARM          CC  Lack of Coping Capacity Index             4.8   

       SurveyYear Indicator Type  INFORMYear  
31499           0    INORM Index        2017  
31500           0    INORM Index        2017  
31501           0    INORM Index        2017  
31502           0    INORM Index        2017  
31503           0    INORM Index        2017  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31499 to 31689
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [27]:
# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df3 in df3s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2017 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df3.to_csv(file_path, index=False)
    print(f"DataFrame for 2017 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2017 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2017 Select INFORM Index - CC.csv
DataFrame for 2017 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2017 Select INFORM Index - CC.INF.csv
DataFrame for 2017 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2017 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2017 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2017 Select INFORM Index - HA.csv
DataFrame for 2017 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2017 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2017 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\

### 2018 INFORM Indeces

In [28]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df4 = df4.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df4s = {}

# Splitting and storing DataFrames based on category
for category, group_df4 in grouped_df4:
    if category in selected_indicator_ids:
        df4s[category] = group_df4.copy()

# Accessing the new DataFrames and printing info
for category, new_df4 in df4s.items():
    print(f"DataFrame of 2018 Indices for Indicator Id {category}:")
    print(new_df4.head())
    print()  # line to space between dataframes
    print(new_df4.info())
    print()  # line to space between dataframes

DataFrame of 2018 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31503  AFG          CC  Lack of Coping Capacity Index             7.3   
31504  AGO          CC  Lack of Coping Capacity Index             7.0   
31505  ARE          CC  Lack of Coping Capacity Index             1.8   
31506  ARG          CC  Lack of Coping Capacity Index             3.4   
31507  ARM          CC  Lack of Coping Capacity Index             4.7   

       SurveyYear Indicator Type  INFORMYear  
31503           0    INORM Index        2018  
31504           0    INORM Index        2018  
31505           0    INORM Index        2018  
31506           0    INORM Index        2018  
31507           0    INORM Index        2018  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31503 to 31693
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [29]:
# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df4 in df4s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2018 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df4.to_csv(file_path, index=False)
    print(f"DataFrame for 2018 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2018 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2018 Select INFORM Index - CC.csv
DataFrame for 2018 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2018 Select INFORM Index - CC.INF.csv
DataFrame for 2018 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2018 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2018 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2018 Select INFORM Index - HA.csv
DataFrame for 2018 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2018 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2018 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\

### 2019 INFORM Indeces

In [30]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df5 = df5.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df5s = {}

# Splitting and storing DataFrames based on category
for category, group_df5 in grouped_df5:
    if category in selected_indicator_ids:
        df5s[category] = group_df5.copy()

# Accessing the new DataFrames and printing info
for category, new_df5 in df5s.items():
    print(f"DataFrame of 2019 Indices for Indicator Id {category}:")
    print(new_df5.head())
    print()  # line to space between dataframes
    print(new_df5.info())
    print()  # line to space between dataframes

DataFrame of 2019 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31523  AFG          CC  Lack of Coping Capacity Index             7.2   
31524  AGO          CC  Lack of Coping Capacity Index             6.9   
31525  ARE          CC  Lack of Coping Capacity Index             1.8   
31526  ARG          CC  Lack of Coping Capacity Index             3.4   
31527  ARM          CC  Lack of Coping Capacity Index             4.7   

       SurveyYear Indicator Type  INFORMYear  
31523           0    INORM Index        2019  
31524           0    INORM Index        2019  
31525           0    INORM Index        2019  
31526           0    INORM Index        2019  
31527           0    INORM Index        2019  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31523 to 31713
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [31]:
# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df5 in df5s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2019 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df5.to_csv(file_path, index=False)
    print(f"DataFrame for 2019 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2019 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2019 Select INFORM Index - CC.csv
DataFrame for 2019 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2019 Select INFORM Index - CC.INF.csv
DataFrame for 2019 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2019 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2019 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2019 Select INFORM Index - HA.csv
DataFrame for 2019 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2019 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2019 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\

### 2020 INFORM Indeces

In [32]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df6 = df6.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df6s = {}

# Splitting and storing DataFrames based on category
for category, group_df6 in grouped_df6:
    if category in selected_indicator_ids:
        df6s[category] = group_df6.copy()

# Accessing the new DataFrames and printing info
for category, new_df6 in df6s.items():
    print(f"DataFrame of 2020 Indices for Indicator Id {category}:")
    print(new_df6.head())
    print()  # line to space between dataframes
    print(new_df6.info())
    print()  # line to space between dataframes

DataFrame of 2020 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31488  AFG          CC  Lack of Coping Capacity Index             7.2   
31489  AGO          CC  Lack of Coping Capacity Index             6.8   
31490  ARE          CC  Lack of Coping Capacity Index             1.7   
31491  ARG          CC  Lack of Coping Capacity Index             3.4   
31492  ARM          CC  Lack of Coping Capacity Index             4.5   

       SurveyYear Indicator Type  INFORMYear  
31488           0    INORM Index        2020  
31489           0    INORM Index        2020  
31490           0    INORM Index        2020  
31491           0    INORM Index        2020  
31492           0    INORM Index        2020  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31488 to 31678
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [33]:
# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df6 in df6s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2020 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df6.to_csv(file_path, index=False)
    print(f"DataFrame for 2020 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2020 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2020 Select INFORM Index - CC.csv
DataFrame for 2020 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2020 Select INFORM Index - CC.INF.csv
DataFrame for 2020 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2020 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2020 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2020 Select INFORM Index - HA.csv
DataFrame for 2020 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2020 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2020 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\

### 2021 INFORM Indeces

In [34]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df7 = df7.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df7s = {}

# Splitting and storing DataFrames based on category
for category, group_df7 in grouped_df7:
    if category in selected_indicator_ids:
        df7s[category] = group_df7.copy()

# Accessing the new DataFrames and printing info
for category, new_df7 in df7s.items():
    print(f"DataFrame of 2021 Indices for Indicator Id {category}:")
    print(new_df7.head())
    print()  # line to space between dataframes
    print(new_df7.info())
    print()  # line to space between dataframes

DataFrame of 2021 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31485  AFG          CC  Lack of Coping Capacity Index             7.1   
31486  AGO          CC  Lack of Coping Capacity Index             6.9   
31487  ARE          CC  Lack of Coping Capacity Index             1.8   
31488  ARG          CC  Lack of Coping Capacity Index             3.5   
31489  ARM          CC  Lack of Coping Capacity Index             4.4   

       SurveyYear Indicator Type  INFORMYear  
31485           0    INORM Index        2021  
31486           0    INORM Index        2021  
31487           0    INORM Index        2021  
31488           0    INORM Index        2021  
31489           0    INORM Index        2021  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31485 to 31675
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [35]:
# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df7 in df7s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2021 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df7.to_csv(file_path, index=False)
    print(f"DataFrame for 2021 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2021 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2021 Select INFORM Index - CC.csv
DataFrame for 2021 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2021 Select INFORM Index - CC.INF.csv
DataFrame for 2021 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2021 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2021 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2021 Select INFORM Index - HA.csv
DataFrame for 2021 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2021 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2021 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\

### 2022 INFORM Indeces

In [36]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df8 = df8.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df8s = {}

# Splitting and storing DataFrames based on category
for category, group_df8 in grouped_df8:
    if category in selected_indicator_ids:
        df8s[category] = group_df8.copy()

# Accessing the new DataFrames and printing info
for category, new_df8 in df8s.items():
    print(f"DataFrame of 2021 Indices for Indicator Id {category}:")
    print(new_df8.head())
    print()  # line to space between dataframes
    print(new_df8.info())
    print()  # line to space between dataframes

DataFrame of 2021 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31501  AFG          CC  Lack of Coping Capacity Index             7.2   
31502  AGO          CC  Lack of Coping Capacity Index             6.8   
31503  ARE          CC  Lack of Coping Capacity Index             1.7   
31504  ARG          CC  Lack of Coping Capacity Index             3.6   
31505  ARM          CC  Lack of Coping Capacity Index             4.4   

       SurveyYear Indicator Type  INFORMYear  
31501           0    INORM Index        2022  
31502           0    INORM Index        2022  
31503           0    INORM Index        2022  
31504           0    INORM Index        2022  
31505           0    INORM Index        2022  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31501 to 31691
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [37]:
# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df8 in df8s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2022 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df8.to_csv(file_path, index=False)
    print(f"DataFrame for 2022 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2022 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2022 Select INFORM Index - CC.csv
DataFrame for 2022 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2022 Select INFORM Index - CC.INF.csv
DataFrame for 2022 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2022 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2022 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2022 Select INFORM Index - HA.csv
DataFrame for 2022 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2022 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2022 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\

### 2023 INFORM Indeces

In [38]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df9 = df9.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df9s = {}

# Splitting and storing DataFrames based on category
for category, group_df9 in grouped_df9:
    if category in selected_indicator_ids:
        df9s[category] = group_df9.copy()

# Accessing the new DataFrames and printing info
for category, new_df9 in df9s.items():
    print(f"DataFrame of 2023 Indices for Indicator Id {category}:")
    print(new_df9.head())
    print()  # line to space between dataframes
    print(new_df9.info())
    print()  # line to space between dataframes

DataFrame of 2023 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31490  AFG          CC  Lack of Coping Capacity Index             7.1   
31491  AGO          CC  Lack of Coping Capacity Index             6.7   
31492  ARE          CC  Lack of Coping Capacity Index             1.8   
31493  ARG          CC  Lack of Coping Capacity Index             3.6   
31494  ARM          CC  Lack of Coping Capacity Index             4.5   

       SurveyYear Indicator Type  INFORMYear  
31490           0    INORM Index        2023  
31491           0    INORM Index        2023  
31492           0    INORM Index        2023  
31493           0    INORM Index        2023  
31494           0    INORM Index        2023  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31490 to 31680
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [39]:
# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df9 in df9s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2023 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df9.to_csv(file_path, index=False)
    print(f"DataFrame for 2023 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2023 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2023 Select INFORM Index - CC.csv
DataFrame for 2023 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2023 Select INFORM Index - CC.INF.csv
DataFrame for 2023 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2023 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2023 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2023 Select INFORM Index - HA.csv
DataFrame for 2023 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2023 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2023 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\

### 2024 INFORM Indeces

In [40]:
# Select 11 specific indicator IDs
selected_indicator_ids = ['INFORM', 'HA', 'HA.NUM', 'HA.NUM.CON', 'HA.NAT.EPI', 'VU', 'VU.SEV', 'VU.VGR', 'CC', 'CC.INF', 
                          'CC.INS.GOV']  # Example list of indicator IDs to select

# Grouping by 'Indicator ID'
grouped_df10 = df10.groupby('IndicatorId')

# Creating a dictionary to store DataFrames
df10s = {}

# Splitting and storing DataFrames based on category
for category, group_df10 in grouped_df10:
    if category in selected_indicator_ids:
        df10s[category] = group_df10.copy()

# Accessing the new DataFrames and printing info
for category, new_df10 in df10s.items():
    print(f"DataFrame of 2024 Indices for Indicator Id {category}:")
    print(new_df10.head())
    print()  # line to space between dataframes
    print(new_df10.info())
    print()  # line to space between dataframes

DataFrame of 2024 Indices for Indicator Id CC:
      Iso3 IndicatorId                  IndicatorName  IndicatorScore  \
31321  AFG          CC  Lack of Coping Capacity Index             7.1   
31322  AGO          CC  Lack of Coping Capacity Index             6.7   
31323  ARE          CC  Lack of Coping Capacity Index             1.8   
31324  ARG          CC  Lack of Coping Capacity Index             3.6   
31325  ARM          CC  Lack of Coping Capacity Index             4.5   

       SurveyYear Indicator Type  INFORMYear  
31321           0    INORM Index        2024  
31322           0    INORM Index        2024  
31323           0    INORM Index        2024  
31324           0    INORM Index        2024  
31325           0    INORM Index        2024  

<class 'pandas.core.frame.DataFrame'>
Index: 191 entries, 31321 to 31511
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Iso3            191 non-null   

In [41]:
# Define the folder path where you want to save the DataFrames
folder_path = '~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id'  # Replace this with your desired folder path

# Accessing the new DataFrames and saving them
for category, new_df10 in df10s.items():
    # Construct the file path
    file_path = os.path.join(folder_path, f"2024 Select INFORM Index - {category}.csv")

    # Save each DataFrame as a separate CSV file in the specified folder
    new_df10.to_csv(file_path, index=False)
    print(f"DataFrame for 2024 Select INFORM Year Indeces for {category} saved as {file_path}")

DataFrame for 2024 Select INFORM Year Indeces for CC saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2024 Select INFORM Index - CC.csv
DataFrame for 2024 Select INFORM Year Indeces for CC.INF saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2024 Select INFORM Index - CC.INF.csv
DataFrame for 2024 Select INFORM Year Indeces for CC.INS.GOV saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2024 Select INFORM Index - CC.INS.GOV.csv
DataFrame for 2024 Select INFORM Year Indeces for HA saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2024 Select INFORM Index - HA.csv
DataFrame for 2024 Select INFORM Year Indeces for HA.NAT.EPI saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\2024 Select INFORM Index - HA.NAT.EPI.csv
DataFrame for 2024 Select INFORM Year Indeces for INFORM saved as ~/Desktop/CodeOp/DSF02/Group Project/2014-2024 Index by Indicator Id\